nucypher/blockchain/eth/actors.py

```diff
@@ -656,3 +696,342 @@ def request_evaluation(self, evidence):
+
+    @classmethod
+    def from_configuration_file(cls, filepath: str = None, **overrides) -> 'StakeHolder':
+        filepath = filepath or cls.default_filepath()
+        payload = cls._read_configuration_file(filepath=filepath)
+
+        # Sub config
+        blockchain_payload = payload.pop('blockchain')
+        blockchain = BlockchainInterface.from_dict(payload=blockchain_payload)
+        payload.update(dict(blockchain=blockchain))
+
+        payload.update(overrides)
+        instance = cls(filepath=filepath, **payload)
+        return instance
+
+    @validate_checksum_address
+    def attach_transacting_power(self, checksum_address: str, password: str = None) -> None:
+        try:
+            transacting_power = self.__transacting_powers[checksum_address]
+        except KeyError:
+            transacting_power = TransactingPower(blockchain=self.blockchain, account=checksum_address)
+            self.__transacting_powers[checksum_address] = transacting_power
+        transacting_power.activate(password=password)
+
+    def to_configuration_file(self, *args, **kwargs) -> str:
+        filepath = super().to_configuration_file(modifier=self.funding_account, *args, **kwargs)
+        return filepath
+
+    def connect(self, blockchain: BlockchainInterface = None) -> None:
+        """Go Online"""
+        if not self.staking_agent:
+            self.staking_agent = StakingEscrowAgent(blockchain=blockchain)
+        if not self.token_agent:
+            self.token_agent = NucypherTokenAgent(blockchain=blockchain)
+        self.blockchain = self.token_agent.blockchain
+
+    #
+    # Account Utilities
+    #
+
+    @property
+    def accounts(self) -> list:
+        return self.__accounts
+
+    def __get_accounts(self) -> None:
+        accounts = self.blockchain.client.accounts
+        self.__accounts.extend(accounts)
+
+    def set_funding_account(self, address: str):
+        self.__funding_account = address
+
+    @property
+    def funding_account(self) -> str:
+        if self.__funding_account is NO_FUNDING_ACCOUNT:
+            raise self.NoFundingAccount
+        return self.__funding_account
+
+    @property
+    def funding_tokens(self) -> NU:
+        nunits = self.token_agent.get_balance(address=self.funding_account)
+        return NU.from_nunits(nunits)
+
+    @property
+    def funding_eth(self) -> Decimal:
+        ethers = self.blockchain.client.get_balance(self.funding_account)
+        return ethers
+
+    #
+    # Staking Utilities
+    #
+
+    def read_onchain_stakes(self, account: str = None) -> None:
+        if account:
+            accounts = [account]
+        else:
+            accounts = self.__accounts
+
+        for account in accounts:
+            stakes = list(self.staking_agent.get_all_stakes(staker_address=account))
+            if stakes:
+                staker = Staker(is_me=True, checksum_address=account, blockchain=self.blockchain)
+                self.__stakers[staker] = staker.stakes
```

Comment:
You're using `Staker` as dict key here, but as far as I can tell, this class is not Hashable. Can this be problematic somehow?

Reply:
Yes. Fixed here: `68426a32076f60be232a5b50b458f5da56d06f31`